src/zrc/api/serializers.py

```diff
@@ -333,6 +334,31 @@ class Meta:
             }
         }
 
+    def _get_resultaat_type(self, resultaat_type_url):
+        if not hasattr(self, '_resultaat_type'):
+            self._resultaat_type = None
+            if resultaat_type_url:
+                Client = import_string(settings.ZDS_CLIENT_CLASS)
+                client = Client.from_url(resultaat_type_url)
+                client.auth = APICredential.get_auth(
+                    resultaat_type_url,
+                    scopes=['zds.scopes.zaaktypes.lezen']
+                )
+                self._resultaat_type = client.request(resultaat_type_url, 'resultaattype')
+        return self._resultaat_type
+
+    def _get_resultaat(self, zaak):
+        if not hasattr(self, '_resultaat'):
+            self._resultaat = None
```

Comment:
I don't really see the benefit of adding the caching here - since the `result` not existing raises an exception anyway and the rest of the validation steps are shortcut because of that.

I'd only set the `_resultaat` attribute if it's succesfully retrieved I think.

Reply:
Removed assigning `None` to  `self._resultaat`